core/server/models/base/index.js

```diff
@@ -19,10 +19,12 @@ var _          = require('lodash'),
     validation = require('../../data/validation'),
     plugins    = require('../plugins'),
     i18n       = require('../../i18n'),
-
     ghostBookshelf,
     proto;
 
+// event connector
+require('./when');
```

Comment:
`listeners` > `when` IMO 👍 

Everything else makes sense and I completely agree atm I think `models/index.js` fits better, but long term I definitely want to get rid of having logic inside of index files.